data-protocols/dsp/dsp-catalog/dsp-catalog-api/src/main/java/org/eclipse/edc/protocol/dsp/catalog/api/controller/DspCatalogApiController.java

```diff
@@ -75,43 +75,62 @@ public DspCatalogApiController(Monitor monitor, ObjectMapper mapper, IdentitySer
 
     @POST
     @Path(CATALOG_REQUEST)
-    public Map<String, Object> getCatalog(JsonObject jsonObject, @HeaderParam(AUTHORIZATION) String token) {
+    public Response getCatalog(JsonObject jsonObject, @HeaderParam(AUTHORIZATION) String token) {
         monitor.debug(() -> "DSP: Incoming catalog request.");
 
         var tokenRepresentation = TokenRepresentation.Builder.newInstance()
                 .token(token)
                 .build();
 
-        var claimToken = identityService.verifyJwtToken(tokenRepresentation, dspCallbackAddress)
-                .orElseThrow(failure -> new AuthenticationFailedException());
+        var verificationResult = identityService.verifyJwtToken(tokenRepresentation, dspCallbackAddress);
+        if (verificationResult.failed()) {
+            return errorResponse(Response.Status.UNAUTHORIZED, UNAUTHORIZED);
+        }
 
         var expanded = jsonLdService.expand(jsonObject);
         if (expanded.failed()) {
-            throw new InvalidRequestException(expanded.getFailureDetail());
+            return errorResponse(Response.Status.BAD_REQUEST, BAD_REQUEST);
         }
+
         var expandedJson = expanded.getContent();
         if (!isOfExpectedType(expandedJson, DSPACE_CATALOG_REQUEST_TYPE)) {
-            throw new InvalidRequestException(format("Request body was not of expected type: %s", DSPACE_CATALOG_REQUEST_TYPE));
+            return errorResponse(Response.Status.BAD_REQUEST, BAD_REQUEST);
         }
 
-        var message = transformerRegistry.transform(expandedJson, CatalogRequestMessage.class)
-                .orElseThrow(failure -> new InvalidRequestException(format("Request body was malformed: %s", failure.getFailureDetail())));
+        var transformResult = transformerRegistry.transform(expandedJson, CatalogRequestMessage.class);
+        if (transformResult.failed()) {
+            return errorResponse(Response.Status.BAD_REQUEST, BAD_REQUEST);
+        }
 
+        var message = transformResult.getContent();
         // set protocol
         message.setProtocol(DATASPACE_PROTOCOL_HTTP);
 
-        var catalog = service.getCatalog(message, claimToken)
-                .orElseThrow(exceptionMapper(Catalog.class));
+        var claimToken = verificationResult.getContent();
 
-        var catalogJson = transformerRegistry.transform(catalog, JsonObject.class)
-                .orElseThrow(failure -> new EdcException(format("Failed to build response: %s", failure.getFailureDetail())));
+        Catalog catalog;
+        try {
+            catalog = service.getCatalog(message, claimToken).getContent();
+        } catch (Exception exception) {
+            var errorCode = UUID.randomUUID();
+            monitor.warning(String.format("DSP negotiation error id %s: %s", errorCode, exception));
```

Comment:
This should state that it is an error returning the catalog, not a negotiation

Reply:
oh, copy & paste, sry :D